pandas/core/strings/accessor.py

```diff
@@ -2317,7 +2318,8 @@ def translate(self, table):
         dtype: object
         """
         result = self._data.array._str_translate(table)
-        return self._wrap_result(result)
+        dtype = object if self._data.dtype == "object" else None
+        return self._wrap_result(result, dtype=dtype)
```

Comment:
Maybe dumb question, but we can't just pass `dtype=self._data.dtype` and have the result dtype be correct in `_wrap_result`?

Reply:
Categoricals act out then, that's a bit confusing to me as well, but I didn't want to focus on this now.